README.md

```diff
@@ -57,89 +53,223 @@ The following code creates a KNN index with fields my_vector1, my_vector2, my_ve
-        },
-        "my_vector2": {
-          "type": "knn_vector"
-        },
-        "my_vector3": {
-        "type": "knn_vector"
-      }
-    }
-  }
 }
 ```
 
 * Indexing sample docs to KNN index
 
 ```
-curl -X POST "localhost:9200/myindex/_doc/1" -H 'Content-Type: application/json' -d'
+PUT /myindex/_doc/2?refresh=true
 {
-"my_vector" : [1.5, 2.5],
-"price":10
+    "my_vector" : [1.5, 2.5],
+    "price":10
 }
-'
+```
+
+* Querying K-Nearest neighbors
 
-curl -X PUT "localhost:9200/myindex/_doc/2" -H 'Content-Type: application/json' -d'
+```
+POST /myindex/_search
 {
-"my_vector" : [2.5, 3.5],
-"price":12
+    "size" : 10,
+    "query": {
+        "knn": {
+            "my_vector": {
+                "vector": [3, 4],
+                "k": 2
+            }
+        }
+    }
 }
-'
+```
+
+## Java Native library usage
+For plugin installations from archive(.zip), it is necessary to ensure ```.so``` file for linux OS and ```.jnilib``` file for Mac OS are present in the java library path. This can be possible by copying .so/.jnilib to either $ES_HOME or by adding manually ```-Djava.library.path=<path_to_lib_files>``` in ```jvm.options``` file
+
+## Settings
+### Index Level Settings
+KNN Index level settings should be provided at the time of the index creation. If the settings are not provided, default values will be use. These settings are static which means they cannot be changed after index creation.
 
+##### index.knn
+This setting indicates whether the index uses the KNN Codec or not. Possible values are *true*, *false*. Default value is *false*.
 
-curl -X PUT "localhost:9200/myindex/_doc/3" -H 'Content-Type: application/json' -d'
+##### index.knn.algo_param.m
+This setting is an HNSW parameter that represents "the number of bi-directional links created for every new element during construction. Reasonable range for M is 2-100. Higher M work better on datasets with high intrinsic dimensionality and/or high recall, while low M work better for datasets with low intrinsic dimensionality and/or low recalls. The parameter also determines the algorithm's memory consumption, which is roughly M * 8-10 bytes per stored element." [nmslib/hnswlib](https://github.com/nmslib/hnswlib/blob/master/ALGO_PARAMS.md) The default value is *16*.
+
+##### index.knn.algo_param.ef_search
+This setting is an HNSW parameter that represents "the size of the dynamic list for the nearest neighbors (used during the search). Higher ef leads to more accurate but slower search." [nmslib/hnswlib](https://github.com/nmslib/hnswlib/blob/master/ALGO_PARAMS.md) The default value is *512*.
+
+##### index.knn.algo_param.ef_construction
+This setting is an HNSW parameter that "the parameter has the same meaning as ef, but controls the index_time/index_accuracy. Bigger ef_construction leads to longer construction, but better index quality." [nmslib/hnswlib](https://github.com/nmslib/hnswlib/blob/master/ALGO_PARAMS.md) The default value is *512*.
+
+##### More Information
```

Comment:
Probably OK to remove this section, given that we just linked to the page three times.

Reply:
Makes sense thanks